datastore/README.md

```diff
@@ -0,0 +1,76 @@
+General-purpose key-value storage.
+The keys are strings, and the values are of any type you want.
+
+> **Note**: This crate is meant to be a utility for the implementation of other crates ; it
+>			does not directly participate in the stack of libp2p.
+
+This crate provides the `Datastore` trait, whose template parameter is the type of the value.
+It is implemented on types that represent a key-value storage.
+The only available implementation for now is `JsonFileDatastore`.
+
+# JSON file datastore
+
+The `JsonFileDatastore` can provide a key-value storage that loads and stores data in a single
+JSON file. It is only available if the value implements the `Serialize`, `DeserializeOwned`
+and `Clone` traits.
+
+The `JsonFileDatastore::new` method will attempt to load existing data from the path you pass
+as parameter. This path is also where the data will be stored. The content of the store is
+flushed on destruction or if you call `flush()`.
+
+```rust
+use datastore::Datastore;
+use datastore::JsonFileDatastore;
+
+let datastore = JsonFileDatastore::<Vec<u8>>::new("/tmp/test.json").unwrap();
+datastore.put("foo".into(), vec![1, 2, 3]);
+datastore.put("bar".into(), vec![0, 255, 127]);
+assert_eq!(datastore.get("foo").unwrap(), &[1, 2, 3]);
+datastore.flush().unwrap();		// optional
+```
+
+# Query
+
+In addition to simple operations such as `get` or `put`, the `Datastore` trait also provides
+a way to perform queries on the key-value storage, using the `query` method.
+
+The struct returned by the `query` method implements the `Stream` trait from `futures`,
+meaning that the result is asynchronous.
+
+> **Note**: For now the `get` and `has` methods are theoretically blocking, but the only
+>			available implementation doesn't do any I/O. Maybe these methods will be made
```

Comment:
This note is really confusing, you imply that it's currently non-blocking (by saying that theoretically it could be blocking) but then say that it could be made asyncronous in the future. Can you make this cllearer?